homeassistant/components/zwave_js/light.py

```diff
@@ -0,0 +1,300 @@
+        await self._async_set_brightness(0, kwargs.get(ATTR_TRANSITION))
+
+    async def _async_set_brightness(
+        self, brightness: Optional[int], transition: Optional[int] = None
+    ) -> None:
+        """Set new brightness to light."""
+        await self._async_set_transition_duration(transition)
+        # Zwave multilevel switches use a range of [0, 99] to control
+        # brightness. Level 255 means to set it to previous value.
+        if brightness is None:
+            brightness = 255
+        else:
+            brightness = byte_to_zwave_brightness(brightness)
+        if self._target_value is not None:
+            await self.info.node.async_set_value(self._target_value, brightness)
+        else:
+            await self.info.node.async_set_value(self.info.primary_value, brightness)
+
+    async def _async_set_transition_duration(
+        self, duration: Optional[int] = None
+    ) -> None:
+        """Set the transition time for the brightness value."""
+        if self._dimming_duration is None:
+            return
+
+        # TODO: setting duration needs to be fixed upstream
+        # https://github.com/zwave-js/node-zwave-js/issues/1321
+        return
+
+        if duration is None:
+            # no transition specified by user, use defaults
+            duration = 7621  # anything over 7620 uses the factory default
+        else:
+            # transition specified by user
+            transition = duration
+            if transition <= 127:
+                duration = int(transition)
+            else:
+                minutes = int(transition / 60)
+                LOGGER.debug(
+                    "Transition rounded to %d minutes for %s",
+                    minutes,
+                    self.entity_id,
+                )
+                duration = minutes + 128
+
+        # only send value if it differs from current
+        # this prevents sending a command for nothing
+        if self._dimming_duration.value != duration:
+            await self.info.node.async_set_value(self._dimming_duration, duration)
+
+    @callback
+    def _calculate_color_values(self) -> None:
+        """Calculate light colors."""
+
+        # RGB support
+        red_val = self.get_zwave_value(
+            "currentColor", CommandClass.SWITCH_COLOR, property_key_name="Red"
+        )
+        green_val = self.get_zwave_value(
+            "currentColor", CommandClass.SWITCH_COLOR, property_key_name="Green"
+        )
+        blue_val = self.get_zwave_value(
+            "currentColor", CommandClass.SWITCH_COLOR, property_key_name="Blue"
+        )
+        if red_val and green_val and blue_val:
+            self._supports_color = True
+            # convert to HS
+            self._hs = color_util.color_RGB_to_hs(
+                red_val.value, green_val.value, blue_val.value
+            )
+
+        # Update color temp limits.
+        min_kelvin_val = self.get_zwave_value(81, CommandClass.CONFIGURATION)
```

Comment:
Configuration parameters are not standardized and can specific to individual devices, so parameter 81 is only valid in this context for a couple of light switches. This should really be part of the "quirks" you've talked about. I don't think the core integration should ever be looking at config parameters for decision logic.

Reply:
You are absolutely right. I basically copied this from the ozw integration, not realizing this is device specific. We need a proper way to handle this.

Reply:
Thinking about this the min_mired and max_mired values are light dependent and should either be retrieved from the actual lights specs or be user configurable. These values will be what the light actually emits at full brightness, e.g. 2700K for the warm light.

Reply:
For now I removed the device specific code from the light implementation.